src/component/framework/collection.js

```diff
@@ -1,18 +1,25 @@
 export class Collection {
 
-  activate(model) {
-    this.models   = model.value;
-    this.element  = model.element;
+  activate(vm) {
+    this.vm = vm;
+  }
 
-    /* @note the collection element sets index on every element, consider the
-     * index property on the schema reserved for the collection element */
-    this.schema = (index) => {
```

Comment:
Instead of a function defined in the activate I moved it to a method on the Collection class